File: java/DaveGame.java
import java.util.Random;
import java.io.Console;

class DaveGame
{
    public static void main(String args[])
    {
        Random generator = new Random();
        int randomNumber = generator.nextInt(10);

        while (true) {
            Console console = System.console();
            String guessString = console.readLine("Guess a number (0-10):");

            int guessInt = Integer.parseInt(guessString);

            if (guessInt > randomNumber) {
                System.out.println("Too high! Try again.\n");
            } else if (guessInt < randomNumber) {
                System.out.println("Too low! Try again.\n");
            } else if (guessInt == randomNumber) {
                System.out.println("You guessed it! Goodbye.\n");
                break;
            } else {
                System.out.println("That's not a valid number.\n");
            }
        }
    }
}

File: php/dave_game.php
<?php
$randomNumber = rand(0, 10);

while (true) {
    fwrite(STDOUT, "Guess a number (0-10):");

    $guessString = trim(fgets(STDIN));
    $guessInt = intval($guessString);

    if ($guessInt > $randomNumber) {
        fwrite(STDOUT, "Too high! Guess again.\n");
    } elseif ($guessInt < $randomNumber) {
        fwrite(STDOUT, "Too low! Guess again.\n");
    } elseif ($guessInt == $randomNumber) {
        fwrite(STDOUT, "You guessed it! Goodbye.\n");
        break;
    } else {
        fwrite("That is not a valid number.\n");
    }
}
?>